_posts/2016-03-10-vamos-falar-de-xcode.md

```diff
@@ -0,0 +1,188 @@
+    02CEB9DA1B6E8EDB00488E8F /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 02CEB9D91B6E8EDB00488E8F /* Assets.xcassets */; };
+    02CEB9DD1B6E8EDB00488E8F /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 02CEB9DB1B6E8EDB00488E8F /* LaunchScreen.storyboard */; };
+    02CEB9E81B6E8EDB00488E8F /* CatsTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 02CEB9E71B6E8EDB00488E8F /* CatsTests.swift */; };
+    02CEB9F31B6E8EDB00488E8F /* CatsUITests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 02CEB9F21B6E8EDB00488E8F /* CatsUITests.swift */; };
+/* End PBXBuildFile section */
+
+/* Begin PBXContainerItemProxy section */
+    02CEB9E41B6E8EDB00488E8F /* PBXContainerItemProxy */ = {
+      isa = PBXContainerItemProxy;
+      containerPortal = 02CEB9C71B6E8EDB00488E8F /* Project object */;
+      proxyType = 1;
+      remoteGlobalIDString = 02CEB9CE1B6E8EDB00488E8F;
+      remoteInfo = Cats;
+    };
+    02CEB9EF1B6E8EDB00488E8F /* PBXContainerItemProxy */ = {
+      isa = PBXContainerItemProxy;
+      containerPortal = 02CEB9C71B6E8EDB00488E8F /* Project object */;
+      proxyType = 1;
+      remoteGlobalIDString = 02CEB9CE1B6E8EDB00488E8F;
+      remoteInfo = Cats;
+    };
+/* End PBXContainerItemProxy section */
+
+~~~
+
+Como podemos ver, a maior parte do arquivo é tomada pela chave _objects_, que nada mais é do que um dicionário, com uma chave hexadecimal - usada para referenciar os objetos do projeto de outras partes - e um valor, que em minhas pesquisas indicam ser sempre um outro dicionário. Para facilitar o entendimento, o Xcode adiciona ainda comentário internos, como por exemplo, `/* AppDelegate.swift in Sources */`. Em geral não vamos editar esse arquivo na mão, mas entendê-lo pode facilitar bastante na hora de resolver um conflito de _merge_ com o uso de sistemas de versionamento de arquivos como _git_.
+
+Vamos voltar ao Xcode e a estrutura de diretórios. Podemos ver que mesmo ao criar um projeto simples (por exemplo, via *File->New Project*), temos dentro da pasta do projeto uma subpasta chamada *project.xcworkspace*. A seguir, descrevo os *workspaces* um pouco melhor.
+
+## _Workspaces_
+
+_Workspaces_ mantêm a referência para projetos/subprojetos, _save states_ (janelas e abas abertas, breakpoints do usuário, etc), localização do produto (_app_, biblioteca estática ou dinámica, binários temporários) e o índice de símbolos (classes, métodos, funções, testes, etc), agrupando isso de forma a facilitar a organização de projetos interrelacionados.
+
+E como é o _workspace_ que registra o índice de símbolos, indexando todos os projetos dentro dele, _code completion_, _Jump to definition_ e outras funcionalidades sensíveis ao contexto/conteúdo funcionam de forma transparente entre os projetos. Então se você adiciona sub-projetos, navegar entre eles é bastante facilitado. Com isso, até mesmo refatorar código é feito entre vários projetos de um mesmo _workspace_.
```

Comment:
`E como é o _workspace_ que registra o índice de símbolos, indexando todos os projetos dentro dele, _code completion_, _Jump to definition_ e outras funcionalidades sensíveis ao contexto/conteúdo funcionam de forma transparente entre os projetos.` - não está faltando alguma coisa entre `conteúdo` e `funcionam` em `ao contexto/conteúdo funcionam de forma`? Li o parágrafo e tive a impressão que tem algo faltando. E no texto, `sub-projetos` não tem hífen, ficando `subprojetos`.